_posts/2023-11-15-Codesgiving-Part-1.md

```diff
@@ -0,0 +1,289 @@
+Have you ever noticed something that needs to be fixed or desirable functionality in a project you use?
+___Consider adding your contributions to open-source!___
+
+## All Contributions Matter
+
+Contributing to open-source can come in many forms and contributions don't need to be gigantic to make an impact.
+Software often involves simplifying complexity.
+Simplification requires many actions beyond solely writing code.
+For example, a short walk outside, a conversation with someone, or a nap can sometimes help us with breakthroughs when it comes to development.
+By the same token, open-source benefits greatly from communications on discussion boards, bug or feature descriptions, or other work that might not be strictly considered "engineering".
+
+## An Open-source Contribution Approach
+
+```mermaid!
+flowchart LR
+    reported["Report\nneed(s)"]
+    verification["Verify or replicate\n the need"]
+    verify_check1{"Able to\nverify?"}
+    solution["Hypothesize\n and apply fix"]
+    verify_check2{"Is it\nresolved?"}
+    resolved["Resolution\nand post-actions"]
+
+    reported --> verification
+    verification --> verify_check1
+    verify_check1 --> |No: gather\nmore details| verification
+    verify_check1 --> |Yes: able\nto verify| solution
+    solution --> verify_check2
+    verify_check2 --> |No: try\nsomething else| solution
+    verify_check2 --> |Yes: issue\nresolved|resolved
+
+    style reported fill:#FEF3C7,stroke:#D97706;
+    style resolved fill:#BBF7D0,stroke:#16A34A;
+```
+
+_The troubleshooting process as a workflow involving looped checks for verifying an issue and validating the solution fixes an issue._
+
+It can feel overwhelming to find a way to contribute to open-source.
+Similar to other software methodology, modularizing your approach can help you progress without being overwhelmed.
+Using a troubleshooting approach like the above can help you break down big challenges into bite-sized chunks.
+Consider each step as a "module" or "section" which needs to be addressed sequentially.
+
+### Embrace a Learning Mindset
+
+Open-source contributions almost always entail learning of some kind.
+Many contributions happen solely in the form of code and text communications which are easily misinterpreted.
+__Assume positive intent__ and accept input from others while upholding your own ideas to share successful contributions together.
+Prepare yourself by intentionally opening your mind to input from others, even if you're sure you're absolutely "right".
+
+### Setting Software Scheduling Expectations
+
+
+<script src="https://cdn.jsdelivr.net/npm/vega@5.22.1"></script>
+<script src="https://cdn.jsdelivr.net/npm/vega-lite@5.6.0"></script>
+<script src="https://cdn.jsdelivr.net/npm/vega-embed@6.21.0"></script>
+<div id="vis"></div>
+
+<script>
+  var spec = {
+    "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
+    "description": "Suggested Open-source Time Allocations",
+    "width": 300,
```

Comment:
I've noticed that the legend text gets ellipsized at this width. Perhaps the entire figure or just the legend could be made wider?

It's not a blocker since you can still hover over the corresponding slice to see a tooltip with the full text (which, I might add, is a cool feature), but I thought I'd mention it anyway.

Reply:
Related, might want to add this for smaller screen widths:

```html
<style>
canvas {
  max-width: 100%;
  height: unset;
}
</style>
```

Or if vega allows `width` to be "auto" (full width), I think that'd be fine. Or a max-width of 300px.

Reply:
Thanks - great points and suggestions! I'll see about improving this.